__tests__/routes/courseRoutes.test.js

```diff
@@ -0,0 +1,182 @@
+const request = require('supertest');
+const express = require('express');
+const router = require('../../routes/courseRoutes'); // Import your router file here
+const connectDb = require('../../__tests__/fixtures/db');
+const makeFakeUser = require('../../__tests__/fixtures/fakeUser');
+const { getFakeCourses, getFakeCoursesByCreator } = require('../../__tests__/fixtures/fakeCourses');
+const mongoose = require('mongoose');
+const { signAccessToken } = require('../../helpers/token');
+const errorCodes = require('../../helpers/errorCodes')
+
+const app = express();
+app.use(express.json());
+app.use('/api/courses', router); // Add your router to the Express app
+
+// Mock Google OAuth2 clientID
+jest.mock('../../config/keys', () => {
+  return {
+    GOOGLE_CLIENT_ID: 'test',
+    TOKEN_SECRET: 'test',
+  };
+});
+
+// Start the Express app on a specific port for testing
+const PORT = 5022; // Choose a port for testing
+const ADMIN_ID = 'srdfet784y2uioejqr';
+const server = app.listen(PORT, () => {
+  console.log(`Express server is running on port ${PORT}`);
+});
+
+let fakeUser = makeFakeUser();
+const fakeCourses = getFakeCourses();
+
+describe('Get all courses for user route', () => {
+
+  let db; // Store the database connection
+
+  beforeAll(async () => {
+    db = await connectDb(); // Connect to the database
+
+    // Insert the fake user into the database
+    await db.collection('courses').insertMany(fakeCourses);
+    await db.collection('users').insertOne(fakeUser);
+    fakeUser = await db.collection('users').findOne({ email: fakeUser.email });
+  });
+
+  it('Returns courses made by a given user', async () => {
+
+    const courses = getFakeCoursesByCreator(fakeUser._id);
+    // Send a get request to the courses endpoint
+    const res = await request(`http://localhost:${PORT}`)
+      .get(`/api/courses/creator/${fakeUser._id}`)
```

Comment:
We should take this up before the next sprint. I have read up on it and we should probably use something like:
`localhost:8888/api/courses?creator=<content-creator._id>`

We'll make do with this until then since it does require a few changes. If I have time I might make a PR with this change in the next few days.